### ISSUE TITLE
The Annotation component is resulting in a cyclic import with the module's index

### ISSUE THREAD

**Issue by davicedraz** [react-native-mapbox-gl/maps#783]
By importing Mapbox from '@react-native-mapbox-gl/maps'

And then using the <Mapbox.PointAnnotation> for example, i always get the following warning:

```bash
WARN Require cycle: node_modules\@react-native-mapbox-gl\maps\javascript\index.js -> node_modules\@react-native-mapbox-gl\maps\javascript\components\annotations\Annotation.js -> node_modules\@react-native-mapbox-gl\maps\javascript\index.js

Require cycles are allowed, but can result in uninitialized values. Consider refactoring to remove the need for a cycle.
```
Researching a little about it, in the following [discussion](https://stackoverflow.com/questions/55664673/require-cycles-are-allowed-but-can-result-in-uninitialized-values-consider-ref) it was advised to remove this cycle, for performance purposes and others: 

*"You use a module A in your code, which imports (maybe via other modules) module B, which in turn imports (maybe via other modules) module A. So it is a chicken-egg problem and because either one has to be the first to be loaded, the other will be undefined on importing the first time.*

So solve the chicken-egg problem in your imports and the warning will be gone.

**Comment by mfazekas** [comment]
@davicedraz thanks for the report, could you make a PR addressing this?!

This 
https://github.com/react-native-mapbox-gl/maps/blob/424f931f880c84c9989bb3947e4aaddcdd2aec59/javascript/index.js#L109-L128

could be moved to utils/animated/index.js perhaps?!

**Comment by davicedraz** [comment]
Done! @mfazekas 

I've created a https://github.com/react-native-mapbox-gl/maps/pull/795. I hope I have minimally contributed (;

**Comment by mfazekas** [comment]
@davicedraz thanks. I've reopened the issue, as ideally, we close it if its fix is merged to master. I've added a `Closes: #783` to your PR. This means that when your PR gets merged this issue will be closed automatically.
